tools/windows/DatadogAgentInstaller/CustomActions/Flare.cs

```diff
@@ -0,0 +1,101 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Windows.Forms;
+using Microsoft.Deployment.WindowsInstaller;
+
+namespace Datadog.CustomActions
+{
+    public class Flare
+    {
+        private readonly IInstallerHttpClient _client;
+        private readonly ISession _session;
+
+        public const string DefaultSite = "datadoghq.com";
+
+        public Flare(
+            IInstallerHttpClient client,
+            ISession session)
+        {
+            _client = client;
+            _session = session;
+        }
+
+        public static Flare DefaultFlare(ISession session)
+        {
+            return new Flare(new InstallerWebClient(), session);
+        }
+
+        public void Send(string email)
+        {
+            var agentVersion = CiInfo.PackageVersion;
+            var apikey = _session["APIKEY"];
+            var wixLogLocation = _session["MsiLogFileLocation"];
+            var log = "";
+            try
+            {
+                if (string.IsNullOrEmpty(wixLogLocation))
+                {
+                    _session.Log("Property MsiLogFileLocation is empty");
+                }
+                else if (File.Exists(wixLogLocation))
+                {
+                    using var fs = new FileStream(wixLogLocation, FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
+                    using var sr = new StreamReader(fs, Encoding.Default);
+                    log = sr.ReadToEnd();
+                }
+                else
+                {
+                    _session.Log($"Log file \"{wixLogLocation}\" does not exist");
+                }
+            }
+            catch (Exception e)
+            {
+                _session.Log($"Error retrieving install logs: {e}");
```

Comment:
Interesting that in some cases you dump whole Exception (like here) but in other its Message field like in SendFlare method